src/octree/factory.rs

```diff
@@ -31,9 +31,7 @@ impl OctreeFactory {
             if !octree_argument.starts_with(prefix) {
                 continue;
             }
-            if let Ok(o) = octree_factory_function(octree_argument) {
-                return Ok(o);
-            }
+           return Ok(octree_factory_function(octree_argument)?);
```

Comment:
Let's only change this to remove the redundancy, as discussed. Otherwise looks good.